src/components/AppProvider/AppProvider.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+import { Movie } from '../../types/Movie';
+
+type Context = {
+  movies: Movie[],
+  setMovies: (newMovies: Movie[]) => void,
+  query: string,
+  setQuery: (newQuery: string) => void,
+  movie: Movie | null,
+  setMovie: (newMovie: Movie | null) => void,
+  isMovieExist: boolean,
+  setIsMovieExist: (newCheck: boolean) => void,
+  isSearch: boolean,
+  setIsSearch: (newStatus: boolean) => void,
+};
+
+export const AppContext = React.createContext<Context>({
+  movies: [],
+  setMovies: () => {},
+  query: '',
+  setQuery: () => {},
```

Comment:
you shouldn't add variables that can be changes on each click to the context
use context only for general values for all App, that can affect on many components and will use everywhere